tests/std/tests/P0896R4_ranges_alg_find/test.cpp

```diff
@@ -2,71 +2,56 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <algorithm>
-#include <array>
 #include <cassert>
 #include <concepts>
 #include <ranges>
 #include <utility>
 
 #include <range_algorithm_support.hpp>
+using namespace std;
+using P = std::pair<int, int>;
 
-constexpr void smoke_test() {
-    using ranges::find, ranges::iterator_t, std::same_as;
-    using P = std::pair<int, int>;
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::find(borrowed<false>{}, 42)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::find(borrowed<true>{}, 42)), int*>);
 
-    // Validate dangling story
-    STATIC_ASSERT(same_as<decltype(find(borrowed<false>{}, 42)), ranges::dangling>);
-    STATIC_ASSERT(same_as<decltype(find(borrowed<true>{}, 42)), int*>);
-
-    std::array<P, 3> const pairs = {{{0, 42}, {2, 42}, {4, 42}}};
-
-    for (auto [value, _] : pairs) {
-        {
-            // Validate range overload [found case]
-            auto result = find(basic_borrowed_range{pairs}, value, get_first);
-            STATIC_ASSERT(same_as<decltype(result), iterator_t<basic_borrowed_range<P const>>>);
-            assert((*result).first == value);
+struct instantiator {
+    static constexpr P haystack[3] = {{0, 42}, {2, 42}, {4, 42}};
+
+    template <ranges::input_range Read>
+    static constexpr void call() {
+        using ranges::find, ranges::iterator_t;
+
+        for (auto [value, _] : haystack) {
+            { // Validate range overload [found case]
+                Read wrapped_input{haystack};
+                auto result = find(wrapped_input, value, get_first);
+                STATIC_ASSERT(same_as<decltype(result), iterator_t<Read>>);

```

Comment:
Shall we update all of these return-type `STATIC_ASSERT`s to type-constraint syntax while we're making a pass? We can even add `const` easily:
```suggestion
                const same_as<iterator_t<Read>> auto result =
                    find(wrapped_input, value, get_first);
```

Reply:
Urgh I wanted to complain in your other PR that you should make a STATIC_ASSERT so that do not have to change the world ;)

I would definitely do it in a separate PR though